src/skmultiflow/prototype/rslvq.py

```diff
@@ -0,0 +1,395 @@
+                    eucl_dis = euclidean_distances(xi.reshape(1, xi.size),
+                                                   prototypes[j]
+                                                   .reshape(1, prototypes[j]
+                                                   .size))
+                    if eucl_dis < best_euclid_incorr:
+                        best_euclid_incorr = eucl_dis
+                        incorr_index = j
+
+            # Update nearest wrong prototype and nearest correct prototype
+            # if correct prototype isn't the nearest
+            if best_euclid_incorr < best_euclid_corr:
+                self._update_prototype(j=corr_index, c_xi=c_xi, xi=xi,
+                                       prototypes=prototypes)
+                self._update_prototype(j=incorr_index, c_xi=c_xi, xi=xi,
+                                       prototypes=prototypes)
+
+    def predict(self, x):
+        """Predict class membership index for each input sample.
+        This function does classification on an array of
+        test vectors X.
+        Parameters
+        ----------
+        x : array-like, shape = [n_samples, n_features]
+        Returns
+        -------
+        C : array, shape = (n_samples)
+            Returns predicted values.
+        """
+        return np.array([self.c_w_[np.array([self._costf(xi, p)
+                                             for p in self.w_]).argmax()]
+                        for xi in x])
+
+    def _costf(self, x, w):
+        d = (x - w)[np.newaxis].T
+        d = d.T.dot(d)
+        return - d / (2 * self.sigma)
+
+    def _p(self, j, e, prototypes, y=None):
+        if y is None:
+            fs = [self._costf(e, w) for w in prototypes]
+        else:
+            fs = [self._costf(e, prototypes[i]) for i in
+                  range(prototypes.shape[0]) if
+                  self.c_w_[i] == y]
+
+        fs_max = np.amax(fs)
+        s = sum([np.math.exp(f - fs_max) for f in fs])
+        o = np.math.exp(
+            self._costf(e, prototypes[j]) - fs_max) / s
+        return o
+
+    def predict_proba(self, X):
+        """ predict_proba
+        Predicts the probability of each sample belonging to each one of the
+        known target_values.
+
+        Parameters
+        ----------
+        X: Numpy.ndarray of shape (n_samples, n_features)
+            A matrix of the samples we want to predict.
+
+        Returns
+        -------
+        numpy.ndarray
+            An array of shape (n_samples, n_features), in which each outer
+            entry is associated with the X entry of the same index. And where
+            the list in index [i] contains len(self.target_values) elements,
+            each of which represents the probability that the i-th sample of
+            X belongs to a certain label.
+        """
+        return 'Not implemented'
+
+    def reset(self):
+        """Reset the model"""
+        self.__init__(gradient_descent=self.gradient_descent)
```

Comment:
This `reset` function will have unexpected behavior if the user sets the original instance to different values than the default for parameters other than `gradient_descent`.

Optionally, you can remove this implementation in order to use the one in `BaseSKMObject`.